wire-ios/Wire-iOS/Sources/Analytics/TrackingManager.swift

```diff
@@ -22,24 +22,51 @@ import WireCommonComponents
 import WireSyncEngine
 
 final class TrackingManager: NSObject, TrackingInterface {
+
     private let flowManagerObserver: NSObjectProtocol
+    private let sessionManager: SessionManager
+
+    init(sessionManager: SessionManager) {
+        self.sessionManager = sessionManager
 
-    private override init() {
         AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
 
-        flowManagerObserver = NotificationCenter.default.addObserver(forName: FlowManager.AVSFlowManagerCreatedNotification, object: nil, queue: OperationQueue.main, using: { _ in
-            AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
-        })
-    }
+        flowManagerObserver = NotificationCenter.default.addObserver(
+            forName: FlowManager.AVSFlowManagerCreatedNotification,
+            object: nil,
+            queue: OperationQueue.main,
+            using: { _ in
+                AVSFlowManager.getInstance()?.setEnableMetrics(!ExtensionSettings.shared.disableAnalyticsSharing)
+            }
+        )
 
-    static let shared = TrackingManager()
+        super.init()
+    }
 
     var disableAnalyticsSharing: Bool {
```

Comment:
I think that this property should be tested if it is not already being tested.

Reply:
Out of the scope of this PR but definitely on my mind to do so.